--- SeTmedia.org	2006-08-20 04:41:55.000000000 +0200
+++ SeTmedia	2006-09-25 22:49:15.000000000 +0200
@@ -4,14 +4,21 @@
   mkdir -p $TMP
 fi
 
+unset ITEM
+if [ "$PKGSRC" = "NFS" ]; then
+  ITEM=" --default-item 3 "
+elif [ "$PKGSRC" = "USB" ]; then
+  ITEM=" --default-item 5 "
+fi
 dialog --backtitle "Select Slackware installation source." \
---title "SOURCE MEDIA SELECTION" --menu \
+--title "SOURCE MEDIA SELECTION" $ITEM --menu \
 "Please select the media from which to install Slackware Linux:" \
-11 70 4 \
+12 70 4 \
 "1" "Install from a Slackware CD or DVD" \
 "2" "Install from a hard drive partition" \
 "3" "Install from NFS (Network File System)" \
 "4" "Install from a pre-mounted directory" \
+"5" "Install from a Slackware USB stick" \
 2> $TMP/media
 if [ ! $? = 0 ]; then
  rm $TMP/media
@@ -29,7 +36,7 @@
 elif [ "$SOURCE_MEDIA" = "4" ]; then
  INSdir 
 elif [ "$SOURCE_MEDIA" = "5" ]; then
- INSfd
+ INSUSB
 fi
 
 # Sorry, this goes the way of the dinosaur...
